### PR TITLE
generateData -> llm + lift methods

### PR DESCRIPTION
**Pull request by anotherjesse**
generateData -> llm
- simplified generate code to just be about LLM conversations
- moves all the logic around prep / parsing / ... to recipe
- ergo - json / html mode is now handled by lift inside recipes
- example of using Zod / schemas to parse json results

llm improvements: 
- streaming
- stop sequences
- support for multiple messages (user -> assistant -> user -> assistant)
- support for putting words into LLMs mouth (forcing it to generate json)
- support for (base64) images

common file input:
- added ability to load images / json data from file input

### CODE REVIEW

File: typescript/packages/common-builder/src/built-in.ts [hunk 1]
```diff
@@ -39,6 +38,27 @@ let fetchDataFactory:
...
+  return streamDataFactory(params);
+}
+
+let streamDataFactory:
+  | NodeFactory<
+      { url: string; options?: RequestInit; result?: any },
+      { pending: boolean; result: any; error: any }
+    >
+  | undefined = undefined;
+
```
Comment (anotherjesse) [inline typescript/packages/common-builder/src/built-in.ts:61]: well well well, look what cursor added without me looking :-/.

File: typescript/packages/common-builder/src/built-in.ts [hunk 2]
```diff
@@ -20,7 +20,6 @@ export function generateData<T>(
 export function fetchData<T>(
   params: Value<{
     url: string;
-    mode?: "json" | "text";
```
Comment (anotherjesse) [inline typescript/packages/common-builder/src/built-in.ts:23]: this shouldn't have been removed

**Comment by anotherjesse** [comment]
todo:

- cleanup schema helpers / patterns (perhaps leverage zod?)
- @seefeldb had mentioned migrating generateData to being a recipe - I used lifts... 
- is there a way to return multiple values from a lift?

```
        const { result: jsonResult } = generateText({
            messages: generateJSONMessages({ jsonprompt }),
            system: "You are a helpful assistant that generates JSON objects for testing.  Respond in JSON",
            stop: '```'
        });
```

replace that with something like:

```
        const { result: jsonResult } = generateText( prep({ jsonprompt }) )
```

perhaps this is the recipe idea @seefeldb had mentioned

**Comment by seefeldb** [comment]
Ah yes, when I meant convert to a recipe, I meant to a combination of recipe and lifted functions.

Lifted functions can `return { multiple, things }`, which you can then use in the recipe as `const { multiple, things } = lifted_function(params)`.

**Comment by anotherjesse** [comment]
cool - it might be nice to do:

```
   const maybeHtml = parse(generateText(prep({ prompt })
```

**Comment by seefeldb** [comment]
Yes, this should work! From a caller's perspective there is no difference between a lifted function or a recipe, btw.

**Review by seefeldb — APPROVED** [review]
ship it! looks good overall and we can iterate over details later.

**Pull request merged**